bpf/cgroup/bpf_cgroup_events.h

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright Authors of Tetragon */
+
+#ifndef _BPF_CGROUP_EVENTS__
+#define _BPF_CGROUP_EVENTS__
+
+#include "bpf_helpers.h"
+#include "bpf_events.h"
+#include "environ_conf.h"
+
+#define EVENT_NOFORCE_SEND 0
+#define EVENT_FORCE_SEND   1
+
+/* This function will send the cgroup events to the ring buffer
+ * if Tetragon is running with log level set to LOG_TRACE_LEVEL.
+ * If the parameter force is set to EVENT_FORCE_SEND, then the
+ * event will be forwarded reguardless of the log level.
+ */
+static inline __attribute__((always_inline)) int
+send_cgrp_event(struct bpf_raw_tracepoint_args *ctx,
+		struct cgroup_tracking_value *cgrp_track, __u64 cgrpid,
+		__u32 op, bool force)
+{
+	pid_t pid;
+	char *path;
+	int zero = 0;
+	uint64_t size;
+	struct task_struct *task;
+	struct execve_map_value *curr;
+	struct msg_cgroup_event *msg;
+	struct tetragon_conf *config;
+
+	/* First check if this message must be sent or not */
+	if (likely(force == EVENT_NOFORCE_SEND)) {
+		config = map_lookup_elem(&tg_conf_map, &zero);
+		if (!config)
+			return 0;
+
+		/* We forward bpf events only under TraceLevel */
+		if (likely(config->loglevel != LOG_TRACE_LEVEL))
```

Comment:
bit of nit, but perhaps you can detect this condition from user space and simply not load the program.

Reply:
We definitely need the BPF programs to store cgroup info and cleanup the maps, but sending events mkdir rmdir was more for debugging logging.

Now since we may move the cgroup name to user space we have to send mkdir the event anyway

Reply:
Just an update here: we send events for debugging but also to detect tetragon cgroup config and assert it at user space!

the bpf programs have to be loaded anyway as they keep track of cgroup info inside the cgroup tracking map, which we will need later to do proper filtering.